src/saltext/vmware/modules/esxi.py

```diff
@@ -845,6 +845,326 @@ def get_ntp_config(
+
+    user_name
+        User to delete on matching ESXi hosts. (required).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.remove_user user_name=foo
+    """
+    log.debug("Running vmware_esxi.remove_user")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for h in hosts:
+            h.configManager.accountManager.RemoveUser(user_name)
+            ret[h.name] = True
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+        vim.fault.UserNotFound,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def get_user(
+    user_name=None,
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Get local user on matching ESXi hosts.
+
+    user_name
+        Filter by this user name (required).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.get_user user_name=foo
+    """
+    log.debug("Running vmware_esxi.get_user")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for h in hosts:
+            users = h.configManager.userDirectory.RetrieveUserGroups(
+                searchStr=user_name,
+                belongsToGroup=None,
+                belongsToUser=None,
+                domain=None,
+                exactMatch=True,
+                findUsers=True,
+                findGroups=True,
+            )
+            for user in users:
+                ret[h.name] = {
```

Comment:
As written this would only return one user per host, correct - `ret[h.name]` is overwritten for each user.

Reply:
oops this is a mistake. I have to correct the signature as below as user_name is mandatory in this api. It is supposed to return the user if it exists on the host.

```
def get_user(
    user_name,

```

```
Filter by this user name (required).
```

Reply:
I made a mistake, too. 

I was thinking that there might be multiple users per host that match when I read this line, but it doesn't read like that when reading the function.

`exact_match=True` passed to `RetrieveUserGroups` ensures that only one user will exist so my comment was off.